### ISSUE TITLE
Permission denied, wrong credentials when using sendgrid-php 6.0.0 on PHP 7.2.3

### ISSUE THREAD

**Issue by fterrag** [sendgrid/sendgrid-php#584]
#### Issue Summary

When attempting to call `$sg->client->mail()->send()->post($mail);` per the Quick Start in README.md, I'm receiving the error "Permission denied, wrong credentials". I have verified that the API key I am using is correct and also have tried creating new API keys.

Where it gets interesting is that if I downgrade the sendgrid-php library to version 5.0.0, I'm able to successfully send an email.

Any thoughts here would be helpful! Thanks

#### Steps to Reproduce

Run the code from the Quick Start in README.md.

#### Technical details

* sendgrid-php Version: master 8db454e4b5674b545c207a162b64202b0d63259b (6.0.0) and 9fb94f6a068a6bd887fed665fc33efcd301db486 (5.0.0)
* PHP Version: 7.2.3 on macOS 10.13.3 installed via Homebrew

**Comment by thedotedge** [comment]
Looks like downstream dependency is properly broken — 2 patches have introduced this regression https://github.com/sendgrid/php-http-client/commit/46dcffa8b1c3ff56acbb78124d9e1e37d70d8c30.

Workaround is to pin `"sendgrid/php-http-client": "3.9.0"`

**Comment by fterrag** [comment]
@thedotedge makes sense. Thanks!

**Comment by thedotedge** [comment]
Correction: it's actually 3.8.0 that works, all subsequent version are broken. I wonder if it even makes sense to depend on `"sendgrid/php-http-client": "~3.7"` if downstream is so unstable.

**Comment by fterrag** [comment]
@thedotedge Pinning to 3.9.0 resulted introduced a notice, but 3.8.0 seems to do the trick. Thanks again.

```
Notice: Object of class SendGrid\Client could not be converted to int in /Users/frank/<removed>/vendor/sendgrid/php-http-client/lib/Client.php on line 269
```

**Comment by thinkingserious** [comment]
@thedotedge @fterrag

Please try php-http-client 3.9.2, that version should be good to go. Thanks!

**Comment by thedotedge** [comment]
@thinkingserious does it work for you? For me every request was returning 401.

**Comment by fterrag** [comment]
@thinkingserious I also receive a 401 when using php-http-client 3.9.2. It appears the problem lies on line 195 of Client.php:

```php
        if (isset($headers)) {
            $headers = array_merge($this->headers, $headers);
        } else {
            $headers = [];
        }
```

Should probably be changed to:

```php
        if (isset($headers)) {
            $headers = array_merge($this->headers, $headers);
        } else {
            $headers = $this->headers;
        }
```

I'm not too familiar with the library, so this might not be the best place to address this issue.

**Comment by thinkingserious** [comment]
@thedotedge,

It was working for me when using `php-http-client` directly. I believe @fterrag found the fix though. When we set `$headers = []` we are overwriting the values coming from `sendgrid-php`. I'm fixing now with 3.9.3.

With Best Regards,

Elmer

**Comment by thinkingserious** [comment]
Thanks again to everyone on this thread for helping get a fix pushed out quickly.

I have verified that this bug is fixed using 3.9.3 in `sendgrid-php`.

Thanks again!

**Comment by phansys** [comment]
Related to #585.
